src/packets.ts

```diff
@@ -0,0 +1,100 @@
+import { EventEmitter } from 'events';
+
+export enum PacketState {
+    // The packet has not been sent yet, it may be queued for later sending
+    Pending = 1,
+    // The packet has been sent over the websocket successfully and we are
+    // waiting for a reply.
+    Sending,
+    // The packet was replied to, and has now been complete.
+    Replied,
+    // The caller has indicated they no longer wish to be notified about this event.
+    Cancelled
+}
+
+const maxInt32 = 0xFFFFFFFF;
+
+/**
+ * A Packet is a data type that can be sent over the wire to Constellation.
+ */
+export class Packet extends EventEmitter {
+    private state: PacketState = PacketState.Pending;
+    private timeout: number;
+    private data: {
+        type: 'method',
+        id: number,
+        method: string,
+        params: { [key: string]: any }
+    };
+
+    constructor(private method: string, private params: { [key: string]: any }) {
+        super();
+        this.data = {
+            id: Math.floor(Math.random() * maxInt32),
+            type: 'method',
+            method,
+            params,
+        };
+    }
+
+    /**
+     * Returns the randomly-assigned numeric ID of the packet.
+     * @return {number}
```

Comment:
For JsDoc, there is no difference. For TypeScript, it always should be lowercase `number` as number types aren't instances of Number. (Yay JavaScript). Soo, whichever :P